src/panels/lovelace/cards/hui-gauge-card.ts

```diff
@@ -1,15 +1,127 @@
-import { html } from "@polymer/polymer/lib/utils/html-tag.js";
-import { PolymerElement } from "@polymer/polymer/polymer-element.js";
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import { HomeAssistant } from "../../../types.js";
+import isValidEntityId from "../../../common/entity/valid_entity_id.js";
+import { fireEvent } from "../../../common/dom/fire_event.js";
 
 import "../../../components/ha-card.js";
 
-import EventsMixin from "../../../mixins/events-mixin.js";
+interface Config extends LovelaceConfig {
+  entity: string;
+  title?: string;
+  unit_of_measurement?: string;
+  min?: number;
+  max?: number;
+  severity?: object;
+}
+
+class HuiGaugeCard extends LitElement implements LovelaceCard {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 1;
+  }
+
+  public setConfig(config) {
+    if (!config || !config.entity) {
+      throw new Error("Invalid card configuration");
+    }
+    if (!isValidEntityId(config.entity)) {
+      throw new Error("Invalid Entity");
+    }
+    this._config = { min: 0, max: 100, ...config };
+  }
+
+  protected render() {
+    if (!this._config) {
+      return html``;
+    }
+    const stateObj = this.hass!.states[this._config.entity];
 
-/*
- * @appliesMixin EventsMixin
- */
-class HuiGaugeCard extends EventsMixin(PolymerElement) {
-  static get template() {
+    return html`
+    ${this.renderStyle()}
+    <ha-card  @click="${this._handleClick}">
+    ${
+      !stateObj
+        ? html`<div class="not-found">Entity not available: ${
+            this._config.entity
+          }</div>`
+        : html`
+          <div class='container'>
+            <div class='gauge-a'></div>
+            <div class='gauge-b'></div>
+            <div class='gauge-c' id='gauge'></div>
+            <div class='gauge-data'>
+              <div id='percent'>${stateObj.state}
+              ${this._config!.unit_of_measurement ||
+                stateObj.attributes.unit_of_measurement ||
+                ""}
+              </div>
+              <div id='title'>${this._config.title}
+
+              </div>
+            </div>
+          </div>
+        `
+    }
+    </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps) {
```

Comment:
Can you add the correct type: `changedProps: PropertyValues`

Reply:
We should make sure that all functions that we declare on a class have their types defined.

Reply:
CC @iantrich (as it's a good thing to do)

Reply:
```
Property 'shouldUpdate' in type 'HuiGaugeCard' is not assignable to the same property in base type 'LitElement'.
  Type '(changedProps: Map<string | number | symbol, unknown>) => boolean | Map<string | number | symbol, unknown>' is not assignable to type '(_changedProperties: Map<string | number | symbol, unknown>) => boolean'.
    Type 'boolean | Map<string | number | symbol, unknown>' is not assignable to type 'boolean'.
      Type 'Map<string | number | symbol, unknown>' is not assignable to type 'boolean'.

```

Reply:
So did you read it? 😛 

It's telling you that you're returning the wrong value. Your method returns either `boolean` or `Map<string | number | symbol, unknown>` (which is the type `PropertyValues`). The method you're overriding is expected a return value of `boolean`. Since they are not compatible, it's flagging an error.

It is not flagging an error if you don't type the parameter, because it is treating your method as an overload, which is incorrect.

Always type your function parameters and return value. And always read errors 😉

Reply:
Gotcha. Sorry. I didn't understand what the error meant there but after you explained it actually does make a lot of sense.